core/src/main/scala/chisel3/SeqUtils.scala

```diff
@@ -113,10 +113,14 @@ private[chisel3] object SeqUtils {
             buildAndOrMultiplexor(sels.zip(inWidthMatched))
           }
 
-        case _: Aggregate =>
+        case agg: Aggregate =>
           val allDefineWidth = in.forall { case (_, element) => element.widthOption.isDefined }
-          if(allDefineWidth) {
-            buildAndOrMultiplexor(in)
+          if (allDefineWidth) {
+            val out = Wire(agg)
+            out.getElements.zipWithIndex.map { case (element, i) =>
+              element := oneHotMux(in.map(_._1) zip in.map(_._2.asInstanceOf[Aggregate].getElements(i)))
+            }
```

Comment:
This could be slow on large Aggregates so I think it'd be nice to hoist some of the repeated iteration:
```suggestion
            val (sel, inData) = in.unzip
            val inElts = inData.map(_.asInstanceOf[Aggregate].getElements)
            out.getElements.zip(inElts).foreach { case (outElt, elts) =>
              outElt := oneHotMux(sel.zip(elts))
            }
```